porcelain/miner_test.go

```diff
@@ -599,3 +600,97 @@ func TestMinerGetLastCommittedSectorID(t *testing.T) {
 
 	assert.Equal(t, int(lastCommittedSectorID), 5432)
 }
+
+type minerSetWorkerAddressPlumbing struct {
+	getOwnerFail, getWorkerFail, msgFail, msgWaitFail, cfgFail bool
+	minerAddr, ownerAddr, workerAddr                           address.Address
+}
+
+func (mswap *minerSetWorkerAddressPlumbing) MessageSend(ctx context.Context, from, to address.Address, value types.AttoFIL, gasPrice types.AttoFIL, gasLimit types.GasUnits, method string, params ...interface{}) (cid.Cid, error) {
+
+	if mswap.msgFail {
+		return cid.Cid{}, errors.New("MsgFail")
+	}
+	return types.EmptyMessagesCID, nil
+}
+
+func (mswap *minerSetWorkerAddressPlumbing) MessageWait(ctx context.Context, msgCid cid.Cid, cb func(*types.Block, *types.SignedMessage, *types.MessageReceipt) error) error {
+	if mswap.msgWaitFail {
+		return errors.New("MsgWaitFail")
+	}
+	return nil
+}
+
+func (mswap *minerSetWorkerAddressPlumbing) ConfigGet(dottedKey string) (interface{}, error) {
+	if mswap.cfgFail {
+		return address.Undef, errors.New("ConfigGet failed")
+	}
+	if dottedKey == "mining.minerAddress" {
+		return mswap.minerAddr, nil
+	}
+	return address.Undef, fmt.Errorf("unknown config %s", dottedKey)
+}
+
+func (mswap *minerSetWorkerAddressPlumbing) MinerGetOwnerAddress(ctx context.Context, minerAddr address.Address) (address.Address, error) {
+	if mswap.getOwnerFail {
+		return address.Undef, errors.New("MinerGetOwnerAddress failed")
+	}
+	return mswap.ownerAddr, nil
+}
+
+func TestMinerSetWorkerAddress(t *testing.T) {
+	tf.UnitTest(t)
+
+	minerOwner := address.TestAddress
+	minerAddr := address.NewForTestGetter()()
+	workerAddr := address.NewForTestGetter()()
+	gprice := types.ZeroAttoFIL
+	glimit := types.NewGasUnits(0)
+
+	t.Run("Calling set worker address sets address", func(t *testing.T) {
+		plumbing := &minerSetWorkerAddressPlumbing{
+			workerAddr: workerAddr,
+			ownerAddr:  minerOwner,
+			minerAddr:  minerAddr,
+		}
+
+		err := MinerSetWorkerAddress(context.Background(), plumbing, workerAddr, gprice, glimit)
+		assert.NoError(t, err)
+		assert.Equal(t, workerAddr.String(), plumbing.workerAddr.String())
+	})
+
+	testCases := []struct {
```

Comment:
this setup & test plumbing with failure flags actually caught a missed error check early on.